.github/workflows/tag.yml

```diff
@@ -0,0 +1,110 @@
+        then
+          echo "::set-env name=BV_PART::major"
+        elif [ $IS_MINOR == 'true' ]
+        then
+          echo "::set-env name=BV_PART::minor"
+        else
+          echo "::set-env name=BV_PART::patch"
+        fi
+    - name: Checkout repository
+      uses: actions/checkout@master
+    - name: Set up git user
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+    - name: Checkout master
+      run: git checkout master
+    - name: Squash and merge PR#${{ github.event.pull_request.number }} to master
+      if: github.event.action == 'labeled'
+      run: |
+        git merge --squash "origin/${GITHUB_HEAD_REF}"
+        git commit -m "${PR_TITLE} (#${PR_NUMBER})"
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install bumpversion
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        pip install bumpversion
+    - name: Run bumpversion ${{ env['BV_PART'] }}
+      run: |
+        OLD_TAG=$(git describe --tags --abbrev=0)
+        echo "::set-env name=OLD_TAG::${OLD_TAG}"
+        bumpversion $BV_PART --message "Bump version: {current_version} → {new_version}
+
+        Triggered by #${PR_NUMBER} via GitHub Actions."
+        NEW_TAG=$(git describe --tags --abbrev=0)
+        echo "::set-env name=NEW_TAG::${NEW_TAG}"
+        git tag -n99 -l $NEW_TAG
+
+        CHANGES=$(git log --pretty=format:'%s' $OLD_TAG..HEAD -i -E --grep='^([a-z]*?):')
+        CHANGES_NEWLINE="$(echo "${CHANGES}" | sed -e 's/^/  - /')"
+        SANITIZED_CHANGES=$(echo "${CHANGES}" | sed -e 's/^/<li>/' -e 's|$|</li>|' -e 's/(#[0-9]\+)//' )
+        echo "::set-env name=CHANGES::${SANITIZED_CHANGES//$'\n'/}"
+        NUM_CHANGES=$(echo -n "$CHANGES" | grep -c '^')
+        echo "::set-env name=NUM_CHANGES::${NUM_CHANGES}"
+        git tag $NEW_TAG $NEW_TAG^{} -f -m "$(printf "This is a $BV_PART release from $OLD_TAG → $NEW_TAG.\n\nChanges:\n$CHANGES_NEWLINE")"
+        git tag -n99 -l $NEW_TAG
+    - name: Comment on issue
+      if: >-
+        github.event.action == 'labeled'
+      uses: actions/github@v1.0.0
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        args: |
+          comment "I've queued this up. When it gets merged, I'll create a ${{ env['BV_PART'] }} release from ${{ env['OLD_TAG'] }} → ${{ env['NEW_TAG'] }} which includes the following ${{ env['NUM_CHANGES'] }} change(s) [including this PR]:<br />${{ env['CHANGES'] }}<br />If you make any more changes, you probably want to re-trigger me again by removing the `bumpversion/${{ env['BV_PART'] }}` label and then adding it back again."
+    - name: Push changes
+      if: >-
+        github.event.action == 'closed' && github.event.pull_request.merged
+      uses: ad-m/github-push-action@v0.5.0
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
When you use the Actions' token, it'll not trigger *any* events at all. Meaning that *none* of the workflows that listen to the `push` event, in particular, will start as a result of this.

Reply:
... what, really? I don't understand how that works. It's a regular `git push` that should always trigger, no matter what? If it doesn't trigger, then... ... huh...

Reply:
Yeah, so I guess this is a documented limitation. Maybe I should make a developer token and use that instead (create a bot for the tag/release) and then I can avoid that limitation...

Reply:
> Yeah, so I guess this is a documented limitation

This is protection from shooting yourself in a leg. Otherwise, it's easy to create a loop of recursively reacting to own events over and over again.
GitHub Apps don't have such limitation but then you're responsible for ignoring events caused by yourself.



> Maybe I should make a developer token

I strongly recommend using a GitHub App token. You'll get access to more APIs with it and the ACL is more granular (accessing everything your account can access vs. just one-repo-related stuff). You can put a fake `localhost` URL there if you don't want to actually deploy a web app.

For a full app deployment, though, it'd be useful for you to skim-through this tutorial: https://tutorial.octomachinery.dev/.

Reply:
Is there a solution that doesn't require using something like Heroku? The whole point here is to have a self-contained solution that doesn't require putting things _outside_ of GitHub Actions...

Reply:
Heroku part is only for receiving events. If you want to just use a token, you can use a fake address like localhost.

Reply:
FYI, I tried to use localhost for the GitHub App -- and it absolutely failed and I could never get a useable token from it. It seems like it provides a secret one uses to generate access tokens (and expires in an hour) and those access tokens aren't that useable as github secrets.

Reply:
For now, I'll use a personal access token. Since I have ACL logic added, I think I'm ok with it.

Reply:
@kratsg yes. Basically, you need to put a private key (PEM file that you get from the GitHub App page) in Secrets and then use it inside of an Aciton to fetch an access token (yes, it has a 1-hour lifetime) using it. It's almost the same as GitHub does when it injects its pre-created tokens — they also have 1-hour lifetime. Maybe even a few minutes less. Wrapper script example: https://github.com/ansible-community/collection_migration/blob/82ca900/gh.py.